[Tze Henn] ip

Superbestron opened the pull request:
# Duke Project
> "Let me handle your tasks for you!"

Here are the things you can do with **Duke**!
* Keeps track of tasks as a list
* Mark tasks as done
* Remove tasks

All you need to do is,
1. Download the JAR [file](https://github.com/Superbestron/ip/releases/tag/A-Jar) and run it 😃 


Some examples of commands that you can do with Duke: `<command> <parameters...>`
```
  todo return book
  deadline assignment 0 /by 2/12/2020 1800
  event carnival /at 12/5/2021 1200
```

Upcoming features:
  - [x] Using Gradle
  - [x] Using Checkstyle
  - [ ] GUI interface

Commit:
Add Gradle support

Superbestron committed:
Implement Greet, Echo, Exit

Superbestron committed:
Implement Add, List

Superbestron committed:
Add Mark as Done Task

Superbestron committed:
Add ToDos, Events, Deadlines

Superbestron committed:
Add Automated Text UI Testing

Superbestron committed:
Add Handle Errors

Superbestron committed:
Add Delete

Superbestron committed:
Fix Handle Errors

Superbestron committed:
Add Enums to represent Duke Commands

Superbestron committed:
Added comments for public functions

Superbestron committed:
Add Save function

Superbestron committed:
Add Dates and Times feature

Superbestron committed:
Merge branch 'branch-Level-8'

Superbestron committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Superbestron committed:
Fix parsing bugs for date and time

Superbestron committed:
Refactor code using more OOP

Superbestron committed:
Add packages

Superbestron committed:
Add Junit tests

Superbestron committed:
Add Find function

Superbestron committed:
Tweak code to comply with NUS's coding standard

Superbestron committed:
Add JavaDocs for most public functions

Superbestron committed:
Merge tag 'A-JavaDoc'

* tag 'A-JavaDoc':
  Add JavaDocs for most public functions

# Conflicts:
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java

Superbestron committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Tweak code to comply with NUS's coding standard

# Conflicts:
#	src/main/java/Duke.java

Superbestron committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

Superbestron committed:
Add Gradle Support

Superbestron committed:
Fix code according to Checkstyle

Superbestron committed:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Fix code according to Checkstyle
  Add Gradle Support

Superbestron committed:
Change file path of Duke.java

Superbestron committed:
Add GUI for Duke

Superbestron committed:
Merge branch 'branch-A-Level-10'

* branch-A-Level-10:
  Add GUI for Duke

bernardwan commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,54 @@
+     * @return A string representing the output.
+     */
+    public String outputFormat() {
+        return " | " + (this.isDone ? "1" : "0") + " | " + this.description;
+    }
+
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
```
Under advanced coding standard, unnecessary use of 'this' should be avoided. 'this' keyword should be used only when a field is shadowed by a method or constructor parameter to avoid unnecessary noise.

bernardwan commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,18 @@
+
+public class Todo extends Task {
+
+    public Todo(String description) {
+        super(description);
+    }
+
+    @Override
+    public String outputFormat() {
+        return "T" + super.outputFormat();
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
```
Should include header comments for all public methods? But maybe not for overrides I'm not too sure.

bernardwan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,214 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        String str;
+        if (input.equals(Commands.BYE.toString().toLowerCase())) {
+            this.isByeCommand = true;
```
Avoid unnecessary use of this with fields.

bernardwan reviewed (commented):
Was struggling to find coding standard violations and had to look at the advanced page so that's good.

Pulpy-y commented on src/main/java/duke/controllers/MainWindow.java:
```diff
@@ -0,0 +1,71 @@
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+
+        String welcomeMessage = "Hello, I'm Duke\n"
```
I feel that the welcome message should not be explicitly declared here. Maybe you can let the ui class handle it?

Pulpy-y commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,214 @@
+        this.isByeCommand = false;
+    }
+
+    public boolean getIsByeCommand() {
```
Maybe you can declare it as 'isByeCommand'.

Pulpy-y commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,214 @@
+        }
+    }
+
+    /** Signals the end of the program */
```
Return type missing in java doc.

Superbestron committed:
Add Assertions to several methods

Use assert feature to document important assumptions that should hold at various points in the code to reduce the occurrences of bugs in the program.

Let's put Assert() methods throughout the code in appropriate places.

This is a quick way to ensure that errors made by us are discovered quickly. For more in-depth testing, JUnit tests will be used.

Superbestron committed:
Merge pull request #1 from Superbestron/branch-A-Assertions

Add Assertions to several methods

Superbestron committed:
Merge branch 'master' of github.com:Superbestron/ip

* 'master' of github.com:Superbestron/ip:
  Add Assertions to several methods

Superbestron committed:
Add ViewSchedule function

The task list of a user could get very long and it might get hard to
view his immediate tasks. Furthermore, the tasks are not sorted
by date and time.

For a chosen date inputted by the user, the user is able to view all his
tasks on that date. The tasks are sorted in chronological order and
separated by hour. For hours which there are no tasks, the hour
header will not be printed out.

This allows the user to easily see the tasks that needs to be done
for each hour.

The command for this is "schedule dd/mm/yyyy".

Superbestron committed:
Merge branch 'branch-B-ViewSchedules'

* branch-B-ViewSchedules:
  Add ViewSchedule function

Superbestron committed:
Refactor code to improve code quality

Some functions were getting too long and it is getting increasingly
hard to read.

Let's extract certain code into separate methods.

This makes code easier to understand and reason.
Hidden bugs become easier to spot as well.

Superbestron committed:
Merge pull request #2 from Superbestron/branch-A-CodeQuality

Refactor code to improve code quality

Superbestron committed:
Merge branch 'master' of github.com:Superbestron/ip

* 'master' of github.com:Superbestron/ip:
  Refactor code to improve code quality

Superbestron committed:
Create gradle.yml

Superbestron committed:
Merge branch 'master' of github.com:Superbestron/ip

* 'master' of github.com:Superbestron/ip:
  Create gradle.yml

Superbestron committed:
Fix Checkstyle Errors

Superbestron committed:
Remove IO redirection tests

Superbestron committed:
Fix DateFormatting Java Bug

Superbestron committed:
Fix Checkstyle Errors

Superbestron committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Fix Checkstyle Errors
  Fix DateFormatting Java Bug
  Remove IO redirection tests

Superbestron committed:
Improve GUI

Superbestron committed:
Update READ.ME

Superbestron committed:
Update README.md

Superbestron committed:
Update README.md

Superbestron committed:
Set theme jekyll-theme-cayman

Superbestron committed:
Update README.md

Superbestron committed:
Merge pull request #4 from Superbestron/readme

Readme

Superbestron committed:
Update README.md

Superbestron committed:
Improve UI

Superbestron committed:
Merge branch 'master' of github.com:Superbestron/ip

* 'master' of github.com:Superbestron/ip:
  Update README.md
  Update README.md
  Set theme jekyll-theme-cayman
  Update README.md
  Update README.md
  Update READ.ME

Superbestron committed:
Fix Checkstyle issues

Superbestron committed:
Update README.md

Superbestron committed:
Merge branch 'master' of github.com:Superbestron/ip

* 'master' of github.com:Superbestron/ip:
  Update README.md

Superbestron committed:
Fix Ui
